src/Content/Version.php

```diff
@@ -502,8 +488,8 @@ public function publish(Language|string $language = 'default'): void
 		// check if publishing is allowed
 		VersionRules::publish($this, $language);
 
-		$latest  = $this->sibling('latest')->read($language);
-		$changes = $this->read($language);
+		$latest  = $this->sibling('latest')->read($language) ?? [];
+		$changes = $this->read($language) ?? [];
```

Comment:
@bastianallgeier this change is necessary as we get otherwise an `foreach argument is null` error. But it makes me wonder whether it is the best decision that `::read()` returns `null`.

Reply:
I wonder about that as well. We still have the ::exists method in addition if you really need to find out if content exists or not.

Reply:
I looked at it again and I think it adds some value and clarity to return null here when the version does not exist, even if it is a bit more effort to write the fallbacks. I would leave it as it is.